File: HyperTimeLogger/Common/Src/Views/HTLLineView.m
//
// Created by Maxim Pervushin on 03/11/15.
// Copyright (c) 2015 Maxim Pervushin. All rights reserved.
//

#import "HTLLineView.h"


@implementation HTLLineView

- (CGFloat)thickness {
    return (CGFloat) (1.0 / [[UIScreen mainScreen] scale]);
}

- (void)setFrame:(CGRect)frame {
    frame.size.height = self.thickness;
    [self setContentHuggingPriority:UILayoutPriorityDefaultLow forAxis:UILayoutConstraintAxisHorizontal];
    [self setContentHuggingPriority:UILayoutPriorityRequired forAxis:UILayoutConstraintAxisVertical];
    [super setFrame:frame];
}

- (CGSize)sizeThatFits:(CGSize)size {
    CGFloat hairline = self.thickness;
    if (size.width > size.height) {
        size.height = hairline;
    } else {
        size.width = hairline;
    }
    return size;
}

- (CGSize)intrinsicContentSize {
    return CGSizeMake(UIViewNoIntrinsicMetric, self.thickness);
}

- (instancetype)initWithFrame:(CGRect)frame {
    self = [super initWithFrame:frame];
    if (self) {
        self.backgroundColor = [UIColor redColor];
    }
    return self;
}

- (void)awakeFromNib {
    [super awakeFromNib];
    self.backgroundColor = [UIColor redColor];
}

@end

File: HyperTimeLogger/Common/Src/Views/HTLLineView.swift
//
//  HTLLineView.swift
//  HyperTimeLogger
//
//  Created by Maxim Pervushin on 28/01/16.
//  Copyright © 2016 Maxim Pervushin. All rights reserved.
//

import UIKit

public class HTLLineView: UIView {

    private var thickness: CGFloat {
        return 1.0 / UIScreen.mainScreen().scale
    }
    
    private var _frame: CGRect
    
    override public var frame: CGRect {
        set {
            setContentHuggingPriority(UILayoutPriorityDefaultLow, forAxis: .Horizontal)
            setContentHuggingPriority(UILayoutPriorityRequired, forAxis: .Vertical)
            _frame = CGRect(x: newValue.origin.x, y: newValue.origin.y, width: newValue.size.width, height: thickness)
            super.frame = _frame
        }
        get {
            return _frame
        }
    }
    
    override public func sizeThatFits(size: CGSize) -> CGSize {
        return CGSizeMake(size.width, thickness)
    }
    
    override public func intrinsicContentSize() -> CGSize {
        return CGSizeMake(UIViewNoIntrinsicMetric, thickness)
    }
    
    override init(frame: CGRect) {
        _frame = CGRectZero
        super.init(frame: frame)
    }
    
    required public init?(coder aDecoder: NSCoder) {
        _frame = CGRectZero
        super.init(coder: aDecoder)
    }
}
